CSharp/BatchExplorer/ViewModels/MainViewModel.cs

```diff
@@ -739,21 +739,84 @@ public CommandBase DeleteSelectedJobs
+                                "Confirm Delete",
+                                MessageBoxButton.YesNo);
+
+                            if (result == MessageBoxResult.Yes)
                             {
-                                System.Threading.Tasks.Task asyncTask = job.DeleteAsync();
-                                AsyncOperationTracker.Instance.AddTrackedOperation(new AsyncOperationModel(
-                                    asyncTask,
-                                    new JobOperation(JobOperation.Delete, job.Id)));
+                                this.LeftSpinnerIsVisible = true;
+                                foreach (var job in this.jobs)
+                                {
+                                    if (job.IsChecked)
+                                    {
+                                        await job.DeleteAsync(false);
+                                    }
+                                }
+
+                                //This is very hacky as the tab count never updated correctly if we refresh
+                                //just after issueing delete async inspite of awaiting for result. Probably it is by default nature
+                                //So keep on refreshing until we see the deleted record inside the refresh list
+                                bool requiredRefresh = true;
+                                while (requiredRefresh)
+                                {                                    
+                                    var refreshedJobs = await dataProvider.GetJobCollectionAsync();
+                                    requiredRefresh = refreshedJobs.Any(a=>deletedIds.Contains(a.Id));
+                                    if(!requiredRefresh)
+                                    {
+                                        this.jobs = new ObservableCollection<JobModel>(refreshedJobs);
+                                        this.jobCollection = CollectionViewSource.GetDefaultView(this.jobs);
+                                        this.selectedJob = null;
+                                        this.Jobs.Refresh();
+
+                                        //Refresh the job pane
+                                        FirePropertyChangedEvent("Jobs");
+                                        FirePropertyChangedEvent("JobTabTitle");
+
+                                        //Refresh the task pane
+                                        FirePropertyChangedEvent("SelectedJob");
+                                        FirePropertyChangedEvent("TasksTabTitle");
+                                    }
+                                    else
+                                    {
+                                        await Task.Delay(1000);
+                                    }
+                                }
+                                
+                                this.LeftSpinnerIsVisible = false;

```

Comment:
You should probably have the spinner visibility code inside of a try/catch/finally block in case anything fails you can guarantee that the spinner will get removed and the user can keep using the panel. You can also popup an error message.